src/app/components/popupSubscribe/DialogComponent.tsx

```diff
@@ -0,0 +1,140 @@
+      handleClickOutside(event);
+    };
+
+    document.addEventListener('mousedown', handleClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleClick);
+    };
+  }, [onClose]);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    const userSubscriptionData: UserSubscrtiptionInterface = {
+      email: email,
+    };
+    if (emailError) {
+      return;
+    } else {
+      const outcome: CRMCreateResponseInterface =
+        await registerSubscriptionData(userSubscriptionData);
+      if (outcome.id) {
+        setSubmissionSuccess(true);
+      }
+    }
+  };
+
+  const [submissionSuccess, setSubmissionSuccess] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState<string | undefined>();
+
+  return (
+    <div ref={popupRef} className={`${styles.dialogBackground}`}>
+      <dialog open={true}>
+        <div className={styles.closeButtonWrapper}>
+          <button onClick={onClose}>
+            <div className={styles.closeButton}>
+              <Image src={CloseButton} alt='Close' title='Close' fill={true} />
+            </div>
+          </button>
+        </div>
+        <form onSubmit={(event) => handleSubmit(event)}>
+          <div className={styles.container}>
+            <div className={styles.imageWrapper}>
+              <Image
+                src={MailboxLady}
+                alt='Subscribe to our Newsletter'
+                title='Subscribe to out Newsletter'
+                fill={true}
+                quality={100}
+              />
+            </div>
+            <div className={styles.contentWrapper}>
+              {!submissionSuccess ? (
+                <>
+                  <div>
+                    <p className={styles.title}>Subscribe to our Newsletter!</p>
+                  </div>
+                  <p className={styles.description}>
+                    Be the first to get exclusive offers and latest news
+                  </p>
+                  <div className={styles.textArea}>
+                    <TextBox
+                      variant={TextboxVariant.LONG}
+                      name={'Email'}
+                      label={''}
+                      type={'email'}
+                      value={email}
+                      required={true}
+                      errorMessage={emailError}
+                      placeholder={'Enter your email address'}
+                      onBlur={() =>
+                        !EMAIL_REGEX.test(email)
+                          ? setEmailError('Email Address is invalid')
+                          : setEmailError(undefined)
+                      }
+                      onChange={(e) => setEmail(e.target.value.trim())}
+                    />
+                  </div>
+                  <div className={styles.buttonArea}>
+                    <ActionButton
+                      type='submit'
+                      label='Subscribe Now'
```

Comment:
Can we use a form and uncontrolled components rather than storing the email in a state variable?
we can also introduce the type email for validations

Reply:
The regex check could be done in the onSubmit function of the form.

Reply:
> we can also introduce the type email for validations

What did you mean by 'type email for validations'?

Reply:
I meant checking against the regex when we trigger onSubmit.